tools/rpm2img

```diff
@@ -408,8 +417,28 @@ mkfs_data() {
   target="${1:?}"
   size="${2:?}"
   offset="${3:?}"
-  mkfs.ext4 -m 0 -d "${DATA_MOUNT}" "${BOTTLEROCKET_DATA}" "${size}"
-  echo "${UNLABELED}" | debugfs -w -f - "${BOTTLEROCKET_DATA}"
+  # Create an XFS filesystem if requested
+  if [ "${XFS_DATA_PARTITION}" == "yes" ] ; then
+    echo "writing XFS filesystem for DATA"
+    # Create a file to write the filesystem to first
+    dd if=/dev/zero of="${BOTTLEROCKET_DATA}" bs=1M count=${size%?}
```

Comment:
nit: `${size}` is known to be non-empty at this point:

```suggestion
    dd if=/dev/zero of="${BOTTLEROCKET_DATA}" bs=1M count=${size}
```

Reply:
This is actually trimming the last character off the `${size}` since `dd` wants a count but the value contains an `M` for `mkfs.ext4`. So you end up with `1024` instead of `1024M`.